src/base/memory_hash.c

```diff
@@ -1194,6 +1350,63 @@ mht_dump (THREAD_ENTRY * thread_p, FILE * out_fp, const MHT_TABLE * ht, const in
   return (cont);
 }
 
+/*
+ * mht_dump_hls - display all entries of hash table for HASH LIST SCAN
+ *   return: TRUE/FALSE
+ *   out_fp(in): FILE stream where to dump; if NULL, stdout
+ *   ht(in): hash table to print
+ *   print_id_opt(in): option for printing hash index vector id
+ *   print_func(in): supplied printing function
+ *   func_args(in): arguments to be passed to print_func
+ *
+ * Note: Dump the header of hash table, and for each entry in hash table,
+ *       call function "print_func" on three arguments: the key of the entry,
+ *       the data associated with the key, and args, in order to print the entry
+ *       print_id_opt - Print hash index ? Will run faster if we do not need to
+ *       print this information
+ */
+int
+mht_dump_hls (THREAD_ENTRY * thread_p, FILE * out_fp, const MHT_HLS_TABLE * ht, const int print_id_opt,
+	      int (*print_func) (THREAD_ENTRY * thread_p, FILE * fp, const void *data, void *args),
+	      void *func_args)
+{
+  HENTRY_HLS_PTR *hvector;		/* Entries of hash table */
+  HENTRY_HLS_PTR hentry;		/* A hash table entry. linked list */
+  unsigned int i;
+  int cont = TRUE;
+
+  assert (ht != NULL);
+
+  if (out_fp == NULL)
+    {
+      out_fp = stdout;
+    }
+
+  fprintf (out_fp,
+	   "HTABLE NAME = %s, SIZE = %d,\n" "NENTRIES = %d, NPREALLOC = %d, NCOLLISIONS = %d\n\n",
+	   ht->name, ht->size, ht->nentries, ht->nprealloc_entries, ht->ncollisions);
+
+  if (print_id_opt)
```

Comment:
"if (print_id_opt == -1)" in **mht_dump** is not necessary?

Reply:
The act related routine has been removed, so there is no need for it.